lib/core/sdam/topology_description.js

```diff
@@ -273,16 +273,23 @@ class TopologyDescription {
   }
 }
 
-function topologyTypeForServerType(serverType, options) {
-  if (serverType === ServerType.Standalone) return TopologyType.Single;
-  if (
-    options.directConnection === false &&
-    (serverType === ServerType.Unknown || serverType === ServerType.RSGhost)
-  ) {
+function topologyTypeForServerType(serverType) {
```

Comment:
Very interesting you don't need to pass `options` in here, you can just derive from `ServerType`.